### PR TITLE
wg_engine: introduce simultaneous flag support for strokes trimming

### PR DESCRIPTION
**Pull request by SergeyLebedkin**
[issues 1479: StrokeTrim](https://github.com/thorvg/thorvg/issues/1479)

image

### CODE REVIEW

**Review by mgrudzinska — CHANGES_REQUESTED** [review]
please handle correctly values < 0
[d.json](https://github.com/user-attachments/files/16378336/d.json)

**Review by mgrudzinska — APPROVED** [review]

**Comment by mgrudzinska** [comment]
@SergeyLebedkin please double check, since introduces some changes https://github.com/thorvg/thorvg/pull/2596.
so you should interpret/wrap the begin/end values inside wg_engine (as I remember (-0.5,0.5) may be a problem

**Comment by SergeyLebedkin** [comment]
> @SergeyLebedkin please double check, since introduces some changes #2596. so you should interpret/wrap the begin/end values inside wg_engine (as I remember (-0.5,0.5) may be a problem

I think that this will make a sense when we will stabilize all rules in #2596. So I will make a changes on this PR by folowing the last changes that will be approved

File: src/renderer/wg_engine/tvgWgRenderData.cpp [hunk 1]
```diff
@@ -291,45 +291,68 @@ void WgRenderDataShape::updateMeshes(WgContext &context, const RenderShape &rsha
...
+    // proceed shapes
+    float totalLen{};
+    for (uint32_t i = 0; i < polylines.count; i++) {
+        totalLen += polylines[i]->len;
+        updateShapes(context, polylines[i]);
+    }
+    // proceed strokes
+    float trimBegin{};
+    float trimEnd{};
+    if (!rshape.stroke->strokeTrim(trimBegin, trimEnd)) { trimBegin = 0.0f; trimEnd = 1.0f; }
```
[the remark this answers is not in the document]
Comment (SergeyLebedkin) [reply src/renderer/wg_engine/tvgWgRenderData.cpp]: fixed

**Review by mgrudzinska — APPROVED** [review]

**Review by hermet — APPROVED** [review]

**Pull request merged**